PWGCF/Tasks/GenericFramework.cxx

```diff
@@ -0,0 +1,192 @@
+    fGFW->AddRegion("refP", 7, pows, 0.4, 0.8, 1, 1);
+    fGFW->AddRegion("full", 7, powsFull, -0.8, 0.8, 1, 2);
+
+    corrconfigs.push_back(fGFW->GetCorrelatorConfig("refP {2} refN {-2}", "ChGap22", kFALSE));
+    corrconfigs.push_back(fGFW->GetCorrelatorConfig("refP {2 2} refN {-2 -2}", "ChGap24", kFALSE));
+    corrconfigs.push_back(fGFW->GetCorrelatorConfig("full {2 -2}", "ChFull22", kFALSE));
+    corrconfigs.push_back(fGFW->GetCorrelatorConfig("full {2 2 -2 -2}", "ChFull24", kFALSE));
+    corrconfigs.push_back(fGFW->GetCorrelatorConfig("refP {3} refN {-3}", "ChGap32", kFALSE));
+    corrconfigs.push_back(fGFW->GetCorrelatorConfig("refP {4} refN {-4}", "ChGap42", kFALSE));
+  }
+
+  void FillFC(const GFW::CorrConfig& corrconf, const double& cent, const double& rndm)
+  {
+    double dnx, val;
+    dnx = fGFW->Calculate(corrconf, 0, kTRUE).Re();
+    if (dnx == 0)
+      return;
+    if (!corrconf.pTDif) {
+      val = fGFW->Calculate(corrconf, 0, kFALSE).Re() / dnx;
+      if (TMath::Abs(val) < 1)
+        fFC->FillProfile(corrconf.Head.Data(), cent, val, 1, rndm);
+      return;
+    }
+    return;
+  }
+
+  void process(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::CentV0Ms>>::iterator const& collision, aod::BCsWithTimestamps const&, myTracks const& tracks)
+  {
+
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+
+    if (cfgAcceptance.value.empty() == false) {
+      cfg.mAcceptance = ccdb->getForTimeStamp<GFWWeights>(cfgAcceptance.value, bc.timestamp());
+      if (cfg.mAcceptance)
+        LOGF(info, "Loaded acceptance histogram from %s (%p)", cfgAcceptance.value.c_str(), (void*)cfg.mAcceptance);
+      else
+        LOGF(warning, "Could not load acceptance histogram from %s (%p)", cfgAcceptance.value.c_str(), (void*)cfg.mAcceptance);
+    }
+
+    if (tracks.size() < 1)
+      return;
+    LOGF(INFO, "Tracks for collision: %d | Vertex: %.1f | INT7: %d | V0M: %.1f", tracks.size(), collision.posZ(), collision.sel7(), collision.centV0M());
+    float vtxz = collision.posZ();
+    registry.fill(HIST("hVtxZ"), vtxz);
+
+    fGFW->Clear();
+    const auto centrality = collision.centV0M();
+    float l_Random = fRndm->Rndm();
+    float weff = 1, wacc = 1;
+
+    for (auto track = tracks.begin(); track != tracks.end(); ++track) {
+      registry.fill(HIST("hPhi"), track.phi());
+      registry.fill(HIST("hEta"), track.eta());
+
+      if (cfg.mEfficiency)
+        weff = cfg.mEfficiency->GetBinContent(cfg.mEfficiency->FindBin(track.pt()));
+      else
+        weff = 1.0;
+      if (weff == 0)
+        continue;
+      weff = 1. / weff;
+      if (cfg.mAcceptance)
+        wacc = cfg.mAcceptance->GetNUA(track.phi(), track.eta(), vtxz);
+      else
+        wacc = 1;
+
+      fGFW->Fill(track.eta(), 1, track.phi(), wacc * weff, 3);
+    }
+    for (unsigned long int l_ind = 0; l_ind < corrconfigs.size(); l_ind++) {
+      FillFC(corrconfigs.at(l_ind), centrality, l_Random);
```

Comment:
The identification of the config for each track is very slow. Better invert the logic, have this as the outer loop. Ideally measure which of the two is faster. We can merge this before the optimization, though

Reply:
You test this later I assume or did you already try?